core/src/components/checkbox/checkbox.scss

```diff
@@ -21,26 +21,42 @@
    * @prop --checkmark-color: Color of the checkbox checkmark when checked
    * @prop --checkmark-width: Stroke width of the checkbox checkmark
    */
-  --background-checked: #{ion-color(primary, base)};
+  --checkbox-background-checked: #{ion-color(primary, base)};
   --border-color-checked: #{ion-color(primary, base)};
   --checkmark-color: #{ion-color(primary, contrast)};
   --checkmark-width: 1;
   --transition: none;
 
   display: inline-block;
+
   position: relative;
+  
+  cursor: pointer;
 
   user-select: none;
   z-index: $z-index-item-input;
 }
 
+// TODO(FW-3100): remove this
+:host(.legacy-checkbox) {
+  width: var(--size);
+  height: var(--size);
+}
+
+// TODO(FW-3100): remove :not selector
+:host(.in-item:not(.legacy-checkbox)) {
```

Comment:
If we reordered the styles, wouldn't that avoid the `:not` selector based on style priority? 

e.g.:

```css
:host(.in-item) {
  width: 100%;
  height: 100%;
}

:host(.legacy-checkbox) {
  width: var(--size);
  height: var(--size);
}
```

Or am I misunderstanding what this style block is used for?

Reply:
Aha, good thinking 👀